syntax = "proto3";

package analyzer;

service AnalyzerService {
  rpc Analyze(AnalyzeRequest) returns (AnalyzeResponse);
}

// Запрос для анализа текста
message AnalyzeRequest {
  string correlation_id = 1;
  string content = 2;
}

// Ответ на запрос анализа текста
message AnalyzeResponse {
  string correlation_id = 1;
  map<string, int32> frequency_distribution = 2;
  repeated Entity entities = 3;
}

message Entity {
  string name = 1;
  string type = 2;
}
